### DISCUSSION TITLE
About Checkstyle

### DISCUSSION THREAD

**Discussion by tcrl** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#60] [Q&A] [answered]
I know that good coding style is part of the grading scheme and that it is checked with Checkstyle, with every 5 warnings from Checkstyle deducting 1%. However, if the warning from Checkstyle is something like "unused import" due to what is provided in the skeleton code, should I remove the import statements despite them not being part of the TODOs, or leave them as is? 

Thanks!

**Comment by Derppening** [comment, marked as answer]
It would be good practice to remove unused imports, so you should do that.

IntelliJ can do that for you as well, see `Code > Optimize Imports`.

**Reply by tcrl** [reply]
Thanks! I also wanted to clarify whether every 5 warnings from Checkstyle deducting 1% means that no marks will be deducted if Checkstyle produces less than 5 warnings from our code?

**Reply by Derppening** [reply]
Yes, but obviously you should still try to make sure there are no errors from CheckStyle.
